vllm/distributed/kv_transfer/kv_connector/v1/shared_storage_connector.py

```diff
@@ -0,0 +1,364 @@
+# SPDX-License-Identifier: Apache-2.0
+import hashlib
+import os
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional
+
+import safetensors
+import torch
+
+from vllm.config import VllmConfig
+from vllm.distributed.kv_transfer.kv_connector.v1.base import (
+    KVConnectorBase_V1, KVConnectorMetadata, KVConnectorRole)
+from vllm.logger import init_logger
+from vllm.v1.core.sched.output import SchedulerOutput
+
+if TYPE_CHECKING:
+    from vllm.attention.backends.abstract import AttentionMetadata
+    from vllm.forward_context import ForwardContext
+    from vllm.v1.core.kv_cache_manager import KVCacheManager
+    from vllm.v1.core.kv_cache_utils import KVCacheBlock
+    from vllm.v1.request import Request
+
+logger = init_logger(__name__)
+
+
+@dataclass
+class ReqMeta:
+    # Request tokens
+    token_ids: torch.Tensor
+    # Slot mappings, should have the same length as token_ids
+    slot_mapping: torch.Tensor
+    # Is store or load
+    is_store: bool
+
+    ## Blocks allocated by the scheduler (no-longer needed)
+    #block_ids: torch.Tensor
+
+    @staticmethod
+    def from_request(request: "Request", block_size: int,
+                     is_store: bool) -> "ReqMeta":
+        valid_num_tokens = align_to_block_size(len(request.prompt_token_ids),
+                                               block_size)
+        token_ids = torch.tensor(request.prompt_token_ids)[:valid_num_tokens]
+        block_ids = torch.tensor(request.block_ids)
+        num_blocks = block_ids.shape[0]
+        block_offsets = torch.arange(0, block_size)
+        slot_mapping = block_offsets.reshape((1, block_size)) + \
+                block_ids.reshape((num_blocks, 1)) * block_size
+        slot_mapping = slot_mapping.flatten()[:valid_num_tokens]
+        return ReqMeta(
+            token_ids=token_ids,
```

Comment:
This `token_ids` could be empty list if the prompt is smaller than block size, but it still generate a empty `ReqMeta`  and append to the `SharedStorageConnectorMetadata#add_request`, it generate a folder `local_storage/d41d8cd98f00b204e9800998ecf8427e` which `d41d8cd98f00b204e9800998ecf8427e` stand for  empty list `[]` of `input_token_ids`, so it would be nice if we can avoid add empty `ReqMeta` into `SharedStorageConnectorMetadata#add_request`. 

Whatever, as this is a debug purpose implementation, this issue can be addressed in another PR.